sale_cancel_reason/models/sale.py

```diff
@@ -28,8 +29,7 @@ class SaleOrder(models.Model):
     cancel_reason_id = fields.Many2one(
         'sale.order.cancel.reason',
         string="Reason for cancellation",
-        readonly=True,
-        ondelete="restrict")
```

Comment:
More common is not to allow this, but to allow reasons to be set to inactive so that they cannot be assigned anymore.